### PR TITLE
Add about ingress

### PR DESCRIPTION
**Pull request by lxpollitt**

### CODE REVIEW

File: about/about-kubernetes-ingress.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+HTTPS requests. The requests are then forwarded via separate connections from the load balancer to the chosen service
+backing pods. As a result, network policy applied to the backing pods can restrict access to only allow connections from the load
+balancer, but cannot restrict access to specific original clients.
+
+### Why use Kubernetes Ingress?
+
+Given that Kubernetes [Services]({{site.baseurl}}/about/about-kubernetes-services) already provide a mechanism for load
+balancing access to services from outside of the cluster, why might you want to use Kubernetes Ingress?
+
+The mainline use case is if you have multiple HTTP / HTTPS services which you want to expose through a single external IP
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:40]: The mainline use case is if you have multiple HTTP / HTTPS services that...

File: about/about-kubernetes-ingress.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+address, perhaps with each service having a different URL path, or perhaps as multiple different domains. This can be a
+lot simpler from a client configuration point of view compared to exposing each service outside of the cluster using
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:42]: reads a bit awkwardly....

Compared to exposing each service outside of the cluster using
Kubernetes Services, which gives each service it's own external IP address, this is lot simpler from a client configuration point of view.

File: about/about-kubernetes-ingress.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+Kubernetes Services which gives each service it's own external IP address.
+
+If on the other hand, your application architecture is fronted by a single "front end" microservice then Kubernetes
+Services likely already meet your needs and you might prefer to not add Ingress to the picture, both from a simplicity
+point of view, and potentially also so you can more easily restrict access to specific clients using network policy.
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:47]: A bit long...maybe this:

On the other hand, if your application architecture is fronted by a single "front end" microservice, then Kubernetes
Services likely already meets your needs; it is simpler than adding Ingress, and allows you to easily restrict access to specific clients using network policy.

File: about/about-kubernetes-ingress.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+In effect, your "front end" microservice already plays the role of Kubernetes Ingress, not that dissimilar to
+[in-cluster ingress](#in-cluster-ingress-solutions) solutions discussed below.
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:49]: Missing some words? 

In effect, your "front end" microservice already plays the role of Kubernetes Ingress, **which is** not that dissimilar to
[in-cluster ingress](#in-cluster-ingress-solutions) solutions discussed below.

File: about/about-kubernetes-ingress.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### Types of Ingress solutions
+
+Broadly speaking there are two types of ingress solutions:
+- In-cluster ingress - where the ingress load balancing is performed by pods within the cluster itself.
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:54]: reduce words for scanning:

- In-cluster ingress - ingress load balancing is performed by pods within the cluster itself.
- External ingress - ingress load balancing is implemented outside of the cluster by appliances or cloud provider capabilities.

File: about/about-kubernetes-ingress.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+In-cluster ingress solutions use software load balancers running in pods within the cluster itself. There are many
+different ingress controllers to consider that follow this pattern, including for example the NGINX ingress controller.
+
+The advantages of this approach are that you can: 
+- horizontally scale your ingress solution up to the limits of Kubernetes
+- choose the ingress controller that best suits your specific needs, for example, with particular load balancing
+  algorithms, or security options.
+
+To get your ingress traffic to the in-cluster ingress pods, the ingress pods are normally exposed externally as a
+Kubernetes service, and any of the standard ways of accessing the service from outside of the cluster can be used. A
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:69]: To get your ingress traffic to the in-cluster ingress pods, the ingress pods are normally exposed externally as a Kubernetes service, so you can use any of the standard ways of accessing the service from outside of the cluster.

File: about/about-kubernetes-ingress.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+common approach is use an external network load balancer, or service IP advertisement, with
+`externalTrafficPolicy:local` to minimize the number of network hops, and to retain client source IP address, which
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:71]: A common approach is to use an external network load balancer, or service IP advertisement so you can use network policy to restrict access to the ingress pods to particular clients using `externalTrafficPolicy:local` to minimize the number of network hops, and retain client source IP address.

File: about/about-kubernetes-ingress.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
+---
+title: About Kubernetes ingress
+description: Learn about Kubernetes ingress!
+---
+
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:5]: Inconsistent use of Ingress and ingress.

File: about/about-kubernetes-ingress.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+![In-cluster ingress]({{site.baseurl}}/images/ingress-in-cluster.svg)
+
+#### External ingress solutions
+
+External ingress solutions use application load balancers outside of the cluster. The exact details and
+features depend on which ingress controller you are using, but most cloud providers include an ingress controller that
+automates the provisioning and management of the cloud provider's application load balancers to provide ingress.
+
+The advantages of this type of ingress solutions are that your cloud provider handles the operational complexity of the
+ingress for you.  The downsides are a potentially more limited set of features compared to the rich range of in-cluster
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:83]: The advantages of these type of ingress solutions are...

File: about/about-kubernetes-ingress.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+of application layer load balancing, which load balances directly to the pods backing each service, without going via
+node-ports or other kube-proxy service handling. This has the advantage of eliminating the potential second network hop
+associated with node ports load balancing to a pod on a different node. The potential disadvantage is that if you are
+operating at very high scales, for example with hundreds of pods backing a service, you may exceed the application layer
+load balancers maximum limit of IPs it can load balance to in this mode. In this case switching to an in-cluster ingress
+solution is likely the better fit for you.
+
+### Show me everything!
+
+All the above diagram focus on connection level (L5-7) representation of ingress and services. You can learn more about
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:103]: All of the above diagram focuses on connection level (L5-7) representation of ingress and services.

File: about/about-kubernetes-ingress.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+the network level (L3-4) interactions involved in handling the connections, including which scenarios client source IP
+addresses are maintained, in the [About Kubernetes Services]({{site.baseurl/about/about-kubernetes-services}}) guide.
+
+If you are already up to speed on how services work under the covers, here's some more complete diagrams show how the details
```
Comment (bmckercher123) [inline about/about-kubernetes-ingress.md:107]: If you are already up to speed on how services work under the covers, here are more complete diagrams that show details of how services are load balanced at the network layer (L3-4).

**Review by bmckercher123 — APPROVED** [review]
Minor nits.

File: about/about-kubernetes-ingress.md (hunk 7 above)
Comment (lxpollitt) [reply about/about-kubernetes-ingress.md:71]: I think I'll leave this one as is. externalTrafficPolicy local is a property of the service, so should be next to the other properties of the service I think.

Comment (lxpollitt) [reply about/about-kubernetes-ingress.md:71]: I have split the sentence into two so it easier to parse though!

File: images/ingress-in-cluster-nlb-local.svg [hunk 12]
```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (stevegaossou) [inline images/ingress-in-cluster-nlb-local.svg:1]: Typo in this diagram, it lists Pod B twice. 

The left-most Pod B (for Service 1) should be Pod A.

File: about/about-kubernetes-ingress.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+
+The advantages of this approach are that you can: 
+- horizontally scale your ingress solution up to the limits of Kubernetes
+- choose the ingress controller that best suits your specific needs, for example, with particular load balancing
+  algorithms, or security options.
+
+To get your ingress traffic to the in-cluster ingress pods, the ingress pods are normally exposed externally as a
+Kubernetes service, so you can use any of the standard ways of accessing the service from outside of the cluster. A
+common approach is use an external network load balancer or service IP advertisement, with `externalTrafficPolicy:local`.
+This minimizes the number of network hops, and the retains client source IP address, which allows network policy to be used
```
Comment (stevegaossou) [inline about/about-kubernetes-ingress.md:71]: Minor: Remove "the" from this section " ... and the retains client source IP address ...".

**Review by stevegaossou — APPROVED** [review]
Found a typo in one of the diagrams that should be fixed and a minor grammar error. 

Other than that LGTM.

**Review by neiljerram — APPROVED** [review]

**Pull request merged**